auth/presentation/src/main/java/com/pronoidsoftware/auth/presentation/register/RegisterScreen.kt

```diff
@@ -0,0 +1,185 @@
+                    R.string.registration_successful,
+                    Toast.LENGTH_LONG,
+                ).show()
+                onSuccessfulRegistration()
+            }
+        }
+    }
+
+    RegisterScreen(
+        state = viewModel.state,
+        onAction = { action ->
+            when (action) {
+                is RegisterAction.OnLoginClick -> onLogInClick()
+                else -> viewModel.onAction(action)
+            }
+        },
+    )
+}
+
+@Composable
+private fun RegisterScreen(state: RegisterState, onAction: (RegisterAction) -> Unit) {
+    val topAppBarState = rememberTopAppBarState()
+    val scrollBehavior = TopAppBarDefaults.enterAlwaysScrollBehavior(
+        state = topAppBarState,
+    )
+    val spacing = LocalSpacing.current
+    TaskyScaffold(
+        topAppBar = {
+            AuthToolbar(
+                title = stringResource(id = R.string.create_account),
+                scrollBehavior = scrollBehavior,
+            )
+        },
+        floatingActionButton = {
+            TaskyFloatingActionButton(
+                icon = BackChevronIcon,
+                onClick = {
+                    onAction(RegisterAction.OnLoginClick)
+                },
+            )
+        },
+    ) { innerPadding ->
+        Column(
+            modifier = Modifier
+                .padding(innerPadding)
+                .fillMaxSize()
+                .verticalScroll(rememberScrollState())
+                .clip(
+                    RoundedCornerShape(
+                        topStart = spacing.authContainerRadius,
+                        topEnd = spacing.authContainerRadius,
+                    ),
+                )
+                .background(
+                    MaterialTheme.colorScheme.background,
+                    shape = RoundedCornerShape(
+                        topStart = spacing.authContainerRadius,
+                        topEnd = spacing.authContainerRadius,
+                    ),
+                )
+                .padding(top = spacing.authPaddingTop)
+                .padding(horizontal = spacing.spaceMedium),
+        ) {
+            TaskyTextField(
+                state = state.name,
+                hint = stringResource(id = R.string.name),
+                endIcon = if (state.isNameValid) {
+                    CheckIcon
+                } else {
+                    null
+                },
+                error = state.name.text.isNotBlank() && !state.isNameValid,
+                modifier = Modifier.fillMaxWidth(),
+            )
+            Spacer(modifier = Modifier.height(spacing.authPaddingInterior))
+            TaskyTextField(
+                state = state.email,
+                hint = stringResource(id = R.string.email),
+                endIcon = if (state.isEmailValid) {
+                    CheckIcon
+                } else {
+                    null
+                },
+                error = state.email.text.isNotBlank() && !state.isEmailValid,
```

Comment:
One issue I found with these TextFieldStates is that they make it a pain to compare state instances. If you look into the `TextFieldState` class, you'll notice it's just a normal class and not a data class which would come with a proper `equals` override.
Just encountered a scenario where I had to assert the initial UI state in a test like this:
```
assertThat(initialState).isEqualTo(LoginState())
```
And it always broke because it compared the text field states by reference and not by content, even though the texts were technically the same. 

I didn't yet find a good solution to this other than overriding the `equals` function of the state again (which is a pain since you'll need to change it each time you change the state), but just something to keep in mind. Google seems to officially recommend putting these states in the ViewModel. Could be an option to exclude them from the state data class, but that also doesn't feel very good.

Reply:
Ah interesting, thanks for the tip! That’s a bummer, but maybe it’ll change since the TextFieldState is still experimental? We can at least hope! So you’re saying that the register state class would need to override the equals function with something like how I've updated the `RegisterState` data class? So that we can compare the text of the TextFieldStates?